### PR TITLE
Handle everything at start of request lifecyle

### PR DESCRIPTION
**Pull request by jsumners**
Same thing as https://github.com/fastify/fastify-cookie/pull/26

### CODE REVIEW

File: plugin.js [hunk 1]
```diff
@@ -81,6 +81,7 @@ function fastifyCachingPlugin (instance, options, next) {
   instance.addHook('onRequest', etagHandleRequest)
   instance.addHook('onSend', etagOnSend)
 
+  instance[Symbol.for('fastify-caching.registered')] = true
```
Comment (mcollina) [inline plugin.js]: Thy this change?

**Review by mcollina — COMMENTED** [review]
LGTM wiht

File: plugin.js (hunk 1 above)
Comment (jsumners) [reply plugin.js]: Primarily so it's easier to test. The module does a couple different types of caching and neither of them are as easy to test for proper registration as the cookie module is.

**Review by mcollina — APPROVED** [review]
LGTM as semver-major

**Pull request merged**